src/main/java/org/sonarlint/intellij/core/ProjectBindingManager.java

```diff
@@ -244,40 +244,20 @@ public Set<ProjectKeyAndBranch> getUniqueProjectKeysAndBranchesPairs() {
     return Set.of();
   }
 
-  public static class ProjectKeyAndBranch {
-    private final String projectKey;
-    private final String branchName;
+  public record ProjectKeyAndBranch(String projectKey, @Nullable String branchName) {
 
-    public ProjectKeyAndBranch(String projectKey, @Nullable String branchName) {
-      this.projectKey = projectKey;
-      this.branchName = branchName;
-    }
-
-    public String getProjectKey() {
-      return projectKey;
-    }
-
-    @CheckForNull
-    public String getBranchName() {
-      return branchName;
-    }
-
-    @Override
-    public boolean equals(Object o) {
-      if (this == o) {
-        return true;
-      }
-      if (o == null || getClass() != o.getClass()) {
-        return false;
+      @Override
+      public boolean equals(Object o) {
+        if (this == o) {
+          return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+          return false;
+        }
+        ProjectKeyAndBranch that = (ProjectKeyAndBranch) o;
+        return projectKey.equals(that.projectKey) && Objects.equals(branchName, that.branchName);
```

Comment:
Interesting, why do we need to keep this overridden `equals` in the record?

Reply:
Probably for no reason 😄 
The equals [implementation](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Record.html#equals(java.lang.Object)) of the record should be the same